paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonBatch.scala

```diff
@@ -28,28 +28,28 @@ import java.util.Objects
 
 /** A Spark [[Batch]] for paimon. */
 case class PaimonBatch(
-    splits: Array[Split],
+    inputPartitions: Array[PaimonInputPartition],
     readBuilder: ReadBuilder,
     metadataColumns: Seq[PaimonMetadataColumn] = Seq.empty)
   extends Batch {
 
   override def planInputPartitions(): Array[InputPartition] =
-    splits.map(new SparkInputPartition(_).asInstanceOf[InputPartition])
+    inputPartitions.map(_.asInstanceOf[InputPartition])
 
   override def createReaderFactory(): PartitionReaderFactory =
     PaimonPartitionReaderFactory(readBuilder, metadataColumns)
 
   override def equals(obj: Any): Boolean = {
     obj match {
       case other: PaimonBatch =>
-        this.splits.sameElements(other.splits) &&
+        this.inputPartitions.sameElements(other.inputPartitions) &&
         readBuilder.equals(other.readBuilder)
 
       case _ => false
     }
   }
 
   override def hashCode(): Int = {
-    Objects.hashCode(splits.toSeq, readBuilder)
```

Comment:
make inputPartitions: `Array[PaimonInputPartition]` to Seq, then we can remove equals and hashcode